libdnf/repo/Repo.cpp

```diff
@@ -1853,28 +1853,29 @@ static const char * lrLogLevelFlagToCStr(GLogLevelFlags logLevelFlag)
     return "USER";
 }
 
+static const Logger::Level lrLogLevelFlagToLevel(GLogLevelFlags logLevelFlag)
+{
+    Logger::Level level = Logger::Level::TRACE;
+    if (logLevelFlag & G_LOG_LEVEL_ERROR)
+        level = Logger::Level::ERROR;
+    else if (logLevelFlag & G_LOG_LEVEL_CRITICAL)
+        level = Logger::Level::CRITICAL;
+    else if (logLevelFlag & G_LOG_LEVEL_WARNING)
+        level = Logger::Level::WARNING;
+    else if (logLevelFlag & G_LOG_LEVEL_MESSAGE)
+        level = Logger::Level::NOTICE;
+    else if (logLevelFlag & G_LOG_LEVEL_INFO)
+        level = Logger::Level::INFO;
+    else if (logLevelFlag & G_LOG_LEVEL_DEBUG)
+        level = Logger::Level::DEBUG;
+    return level;
+}
+
 static void librepoLogCB(G_GNUC_UNUSED const gchar *log_domain, GLogLevelFlags log_level,
                          const char *msg, gpointer user_data) noexcept
 {
-    // Ignore exception during logging. Eg. exception generated during logging of exception is not good.
-    try {
-        auto data = static_cast<LrHandleLogData *>(user_data);
-        auto now = time(NULL);
-        struct tm nowTm;
-        gmtime_r(&now, &nowTm);
-
-        std::ostringstream ss;
-        ss << std::setfill('0');
-        ss << std::setw(4) << nowTm.tm_year+1900 << "-" << std::setw(2) << nowTm.tm_mon+1;
-        ss << "-" << std::setw(2) << nowTm.tm_mday;
-        ss << "T" << std::setw(2) << nowTm.tm_hour << ":" << std::setw(2) << nowTm.tm_min;
-        ss << ":" << std::setw(2) << nowTm.tm_sec << "Z ";
-        ss << lrLogLevelFlagToCStr(log_level) << " " << msg << std::endl;
-        auto str = ss.str();
-        fwrite(str.c_str(), sizeof(char), str.length(), data->fd);
-        fflush(data->fd);
```

Comment:
The format of the logged messages in `librepo.log` will certainly change? Not sure if this should go into some changelog or release notes... it's minor but it's a change.

Reply:
I don't see any difference between the formats.